src/common/interfaces/IGateway.sol

```diff
@@ -111,6 +118,9 @@ interface IGateway is IMessageHandler, IMessageSender, IRecoverable {
     /// @notice Pay upfront to later be able to subsidize messages associated to a pool
     function subsidizePool(PoolId poolId) external payable;
```

Comment:
Maybe a bit cleaner: `depositSubsidy` and `withdrawSubsidy`?

Reply:
I like it!